src/hashstore/hashstore.py

```diff
@@ -1,71 +1,8 @@
-"""Core module for hashstore"""
-import io
-import shutil
-import threading
-import time
-import hashlib
+"""Hashstore Interface"""
+from abc import ABC, abstractmethod
 import importlib.metadata
-import os
-from pathlib import Path
-from contextlib import closing
-from tempfile import NamedTemporaryFile
-from collections import namedtuple
-from hashstore.hashstore_interface import HashStoreInterface
-from hashstore.hashstore_config import (
-    DIR_DEPTH,
-    DIR_WIDTH,
-    SYSMETA_NS,
-    DEFAULT_ALGO_LIST,
-    OTHER_ALGO_LIST,
-)
 
-
-class HashStoreFactory:
-    """A factory class for creating `HashStore`-like objects (classes
-    that implement the 'hashstore_interface' abstract methods)
-
-    This factory class provides a method to retrieve a `hashstore` object
-    based on the specified store type (ex. FileHashStore). It supports the
-    creation of different types of hash stores by mapping store types to
-    specific implementations.
-    """
-
-    def __init__(self):
-        """Initialize the HashStoreFactory with default config values"""
-        self.dir_depth = DIR_DEPTH
-        self.dir_width = DIR_WIDTH
-        self.sysmeta_ns = SYSMETA_NS
-        self.default_algo_list = DEFAULT_ALGO_LIST
-        self.other_algo_list = OTHER_ALGO_LIST
-
-    def get_hashstore(self, store_path, hashstore_type):
-        """Get a `HashStore`-like object based on the specified store type.
-
-        Args:
-            root (str): The root directory for the hash store.
-            hashstore_type (str): The type of `HashStore` to retrieve.
-
-        Returns:
-            HashStore: A hash store object based on the given store type.
-
-        Raises:
-            ValueError: If the given store_type is not supported.
-        """
-        hashstore_type.lower()
-        if hashstore_type == "filehashstore":
-            return FileHashStore(
-                root=store_path,
-                depth=self.dir_depth,
-                width=self.dir_width,
-                sysmeta_ns=self.sysmeta_ns,
-                default_algo_list=self.default_algo_list,
-                other_algo_list=self.other_algo_list,
-            )
-        else:
-            raise ValueError(f"hashstore_type: {hashstore_type} is not supported.")
-
-
-class HashStore:
+class HashStore(ABC):
```

Comment:
pycharm tells me PEP 8 expected 2 blank lines above this one

`PEP 8: E302 expected 2 blank lines, found 1`